app/utils/encryptable_transfer.rb

```diff
@@ -28,6 +30,48 @@ def receive(encryptable, private_key, personal_team_password)
 
   private
 
+  def encryptable_destination_name(encryptable_name, receiver)
+    inbox_folder_names = receiver.inbox_folder.encryptables.pluck(:name)
+    counter = 0
+
+    loop do
+      matching_inbox_names = find_matching_inbox_names(encryptable_name, inbox_folder_names)
+      break if matching_inbox_names.empty?
+
+      # rubocop:disable Metrics/LineLength
+      new_encryptable_name = generate_new_encryptable_name(encryptable_name, counter, matching_inbox_names)
+      # rubocop:enable Metrics/LineLength
+
+      encryptable_name = new_encryptable_name
+      counter += 1
+    end
+    encryptable_name
+  end
+
+  def generate_new_encryptable_name(encryptable_name, counter, matching_inbox_names)
+    clip_regex = / \(\d+\)/
+
+
+    matching_inbox_names.each do |inbox_encryptable_name|
+      if clip_regex.match(inbox_encryptable_name) && clip_regex.match(encryptable_name)
+        encryptable_name = encryptable_name.slice(0..-4)
+        encryptable_name = "#{encryptable_name}(#{counter + 1})"
+      else
+        encryptable_name = "#{encryptable_name} (#{counter + 1})"
+      end
+    end
+
+    encryptable_name
+  end
+
+  def find_matching_inbox_names(new_encryptable_name, inbox_folder_names)
```

Comment:
```suggestion
  def find_matching_inbox_names(new_encryptable_name, inbox_folder_names)
```

better than matching, maybe existing_names?